drivers/gpio/gpio_gecko.c

```diff
@@ -286,22 +270,27 @@ static int gpio_gecko_pin_interrupt_configure(struct device *dev,
 			return -EBUSY;
 		}
 
-		if (((flags & GPIO_INT_LEVELS_LOGICAL) != 0) &&
-		    ((data->general.invert & BIT(pin)) != 0)) {
-			rising_edge = flags & GPIO_INT_LOW_0;
-			falling_edge = flags & GPIO_INT_HIGH_1;
-		} else {
-			rising_edge = flags & GPIO_INT_HIGH_1;
-			falling_edge = flags & GPIO_INT_LOW_0;
-		}
+		bool rising_edge = true;
+		bool falling_edge = true;
+
+		if (trig == GPIO_INT_TRIG_LOW) {
+			rising_edge = false;
+			falling_edge = true;
+		} else if (trig == GPIO_INT_TRIG_HIGH) {
+			rising_edge = true;
+			falling_edge = false;
+		} /* default is GPIO_INT_TRIG_BOTH */
 
 		GPIO_IntConfig(config->gpio_index, pin,
 			       rising_edge, falling_edge, true);
-	} else {
-		GPIO_IntDisable(BIT(pin));
 	}
 
-	WRITE_BIT(data->pin_callback_enables, pin, flags & GPIO_INT_ENABLE);
+	bool en = true;
+	if (mode == GPIO_INT_DISABLE) {
+		en = false;
+	}
+
+	WRITE_BIT(data->pin_callback_enables, pin, en);
```

Comment:
This works well but can be written simpler as
```
	WRITE_BIT(data->pin_callback_enables, pin, mode != GPIO_INT_DISABLE);

```

Reply:
Done.